### PR TITLE
Interactive SVG background: process whole SVG group for state

### PR DESCRIPTION
**Pull request by stefan-hoehn**
Improvement for interactive SVG backgrounds:

Allows to flash and handle state on whole SVG groups without a proxy by using the group's path elements instead, e.g. this whole group can be used directly 

https://github.com/user-attachments/assets/8cc17cb2-fabb-4473-93fb-c686776377cb

### CODE REVIEW

File: bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js [hunk 1]
```diff
@@ -145,6 +145,22 @@ export default {
      * @param {HTMLElement} el
      */
     svgOnMouseOver (el) {
+      function flashTheElement (flashElement, fillColor) {
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:148]:
```suggestion
      function flashElement (el, fillColor) {
```

File: bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js [hunk 2]
```diff
@@ -161,18 +177,14 @@ export default {
...
-            const oldOpacity = flashElement.style.opacity
-            flashElement.style.setProperty(attributeName, fillColor)
-            flashElement.style.opacity = 1
-            flashElement.flashing = true
-            setTimeout(() => {
-              flashElement.style.setProperty(attributeName, oldFill)
-              flashElement.style.opacity = oldOpacity
-              flashElement.flashing = false
-            }, 200)
+          if (flashElement) {
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:180]:
```suggestion
          if (el.querySelector('[flash]')) {
```

File: bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js [hunk 3]
```diff
@@ -161,18 +177,14 @@ export default {
           }, 200)
         } else { // groups cannot be filled, so we need to fill special element marked as "flash"
           const flashElement = el.querySelector('[flash]')
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:179]:
```suggestion
```

File: bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js [hunk 4]
```diff
@@ -161,18 +177,14 @@ export default {
...
-              flashElement.style.opacity = oldOpacity
-              flashElement.flashing = false
-            }, 200)
+          if (flashElement) {
+            flashTheElement(flashElement, fillColor)
+          } else {
+            // let's try flashing all path elements in the group
+            const flashElements = el.querySelectorAll('path')
+            for (const path of flashElements) {
+              flashTheElement(path, fillColor)
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:186]:
```suggestion
              flashElement(path, fillColor)
```

**Review by florian-h05 — COMMENTED** [review]
A few suggestions and one question, other then this looks good.

File: bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js [hunk 5]
```diff
@@ -292,100 +304,107 @@ export default {
...
+        // currently no distinction is made regarding different state-types, only the following are supported (yet)
+        case 'OpenClosed':
+        case 'Percent':
+        case 'HSB':
+        case 'OnOff':
+          const useProxy = tagName === 'g' && svgElementConfig.useProxyElementForState // if proxy should be used and element is of type group
+          const element = (useProxy) ? svgElement.querySelector('[flash]') : svgElement
+          if (element) {
+            processState(useProxy, element)
+          } else {
```
[the remark this answers is not in the document]
Comment (stefan-hoehn) [reply bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:403]: As discussed offline, this is correct as it is.

[the remark this answers is not in the document]
Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:403]: We have discussed something else, this hasn't been discussed yet.

[the remark this answers is not in the document]
Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js:403]: Nevermind is correct.

**Review by florian-h05 — APPROVED** [review]

**Pull request merged**